packages/flutter_tools/lib/src/native_assets.dart

```diff
@@ -174,9 +180,12 @@ class NativeAssetsBuildRunnerImpl implements NativeAssetsBuildRunner {
     if (globals.platform.isWindows) {
       return cCompilerConfigWindows();
     }
-    throwToolExit(
-      'Native assets feature not yet implemented for Android.',
-    );
+    throwToolExit('Should use ndkCCompilerConfig for Android.');
```

Comment:
Shouldn't this be checking for Android, with a more generic fallback, so it's not weirdly wrong if we have another platform in the future?